app/code/core/Mage/Api2/Model/Auth/User/Admin.php

```diff
@@ -63,7 +63,7 @@ public function getRole()
             /** @var Mage_Api2_Model_Acl_Global_Role $role */
             $role = $collection->getFirstItem();
             if (!$role->getId()) {
-                throw new Exception('Admin role not found');
+                throw new Exception('Admin role for user ID ' . $this->getUserId() . ' not found');
```

Comment:
I'd add also the role id no?

Reply:
There is no role ID if I get the code correctly. This exception fires when there's no role found for the user ID.